Sources/StructuralCore/Structural.swift

```diff
@@ -21,103 +74,187 @@ public protocol Structural {
+/// The structural representation of a Swift enum case.
+///
+/// `StructuralCase<...>` is the compiler-synthesized
+/// `StructuralRepresentation` for cases of enums conforming to `Structural`.
+///
+/// - Parameter RawValue: the type of the raw value of the enum case.
+/// - Parameter AssociatedValues: the type representing the names and associated
+///   values of the enum case.
 public struct StructuralCase<RawValue, AssociatedValues> {
-    public var name: StaticString
+    /// The name of the enum case.
+    public var name: String
+    /// The raw value of the enum case.
     public var rawValue: RawValue
+    /// The names and associated values of the enum case.
     public var associatedValues: AssociatedValues
 
+    /// Creates a representation from the given raw value and associated values.
     public init(_ rawValue: RawValue, _ associatedValues: AssociatedValues) {
         self.name = ""
         self.rawValue = rawValue
         self.associatedValues = associatedValues
     }
 
-    public init(_ name: StaticString, _ rawValue: RawValue, _ associatedValues: AssociatedValues) {
+    /// Creates a representation from the given raw value and associated values.
+    public init(
+        _ name: String, _ rawValue: RawValue, _ associatedValues: AssociatedValues
+    ) {
         self.name = name
         self.rawValue = rawValue
         self.associatedValues = associatedValues
     }
 }
 
-/// Structural representation of a heterogeneous list cons cell.
+/// The structural representation of a chain of properties or associated values.
+///
+/// `StructuralCons<...>` is the compiler-synthesized
+/// `StructuralRepresentation` for chaining struct stored properties and enum
+/// case associated values.
+///
+/// - Parameter Value: the type of the first value.
+/// - Parameter Next: the type of the second value.
+///
+/// - Note: `StructuralCons<...>` is used as a type-level heterogeneous cons
+///   list, which is verbose and not particularly human-readable. If variadiac
+///   generics are added to the Swift language, `StructuralCons` can be
+///   deprecated in favor of a less-verbose heterogeneous list representaiton
+///   using variadic generics.
 public struct StructuralCons<Value, Next> {
+    /// The first value.
     public var value: Value
+    /// The next value.
     public var next: Next
 
+    /// Creates a representation from the given values.
     public init(_ value: Value, _ next: Next) {
         self.value = value
         self.next = next
     }
 }
 
-/// Structural representation of an empty heterogeneous list cons cell.
+/// The structural representation of an empty list.
+///
+/// - Note: `StructuralEmpty<...>` is used as the base case of the type-level
+///   heterogeneous cons list `StructuralCons`. If variadiac generics are added
+///   to the Swift language, `StructuralCons` can be deprecated in favor of a
+///   less-verbose heterogeneous list representation using variadic generics.
```

Comment:
I'm not sure if mentioning the variadic generics is relavant, given that there is no concrete proposal at the moment (E.g.: what if variadic generics are compiled down to heterogeneous lists?).

Reply:
Removing mentions of variadic generics sounds good to me! I removed them for now.

> (E.g.: what if variadic generics are compiled down to heterogeneous lists?).

Minor comment: if variadic generics are added, I think what would matter is their user-facing API (enabling a simpler `StructuralStruct<...>` design), not how they are compiled down.